lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java

```diff
@@ -212,6 +261,40 @@ public void onXMLContent(ICompletionRequest request, ICompletionResponse respons
 		}
 	}
 
+	private InsertionStrategy computeGAVInsertionStrategy(ICompletionRequest request) {
```

Comment:
This is so nice, I really like this abstraction.

Reply:
It's inspired from the "strategy" design pattern, but it's not as good as the actual pattern ;)

Reply:
I like it a lot regardless :)